apps/ima/fields/index.js

```diff
@@ -2,9 +2,243 @@
+    formatter: ['ukPostcode'],
+    includeInSummary: false,
+    className: ['govuk-input', 'govuk-input--width-10'],
+    validationLink: {
+      field: 'has-address',
+      value: 'no'
+    }
+  },
+  'immigration-adviser-details': {
+    isPageHeading: true
+  },
+  'legal-representative-phone-number': {
+    labelClassName: 'bold',
+    validate: ['required', 'internationalPhoneNumber', {type: 'maxlength', arguments: [200]}],
+    includeInSummary: false,
+    className: ['govuk-input', 'govuk-!-width-two-thirds']
+  },
+  'legal-representative-fullname': {
+    labelClassName: 'bold',
+    validate: ['required', 'notUrl', {type: 'maxlength', arguments: [200]}],
+    includeInSummary: false,
+    className: ['govuk-input', 'govuk-!-width-two-thirds']
+  },
+  'legal-representative-organisation': {
+    labelClassName: 'bold',
+    validate: ['required', 'notUrl', {type: 'maxlength', arguments: [200]}],
+    includeInSummary: false,
+    className: ['govuk-input', 'govuk-!-width-two-thirds']
+  },
+  'legal-representative-house-number': {
+    validate: ['required', 'notUrl', { type: 'maxlength', arguments: 200 }],
+    includeInSummary: false
+  },
+  'legal-representative-street': {
+    validate: ['notUrl', { type: 'maxlength', arguments: 200 }],
+    includeInSummary: false
+  },
+  'legal-representative-townOrCity': {
+    validate: ['required', 'notUrl',
+      { type: 'regex', arguments: /^([^0-9]*)$/ },
+      { type: 'maxlength', arguments: 200 }
+    ],
+    includeInSummary: false,
+    className: ['govuk-input', 'govuk-!-width-two-thirds']
+  },
+  'legal-representative-county': {
+    validate: ['notUrl',
+      { type: 'regex', arguments: /^([^0-9]*)$/ },
+      { type: 'maxlength', arguments: 200 }
+    ],
+    includeInSummary: false,
+    className: ['govuk-input', 'govuk-!-width-two-thirds']
+  },
+  'legal-representative-postcode': {
+    validate: ['required', 'postcode', {type: 'maxlength', arguments: [200]}],
+    formatter: ['ukPostcode'],
+    includeInSummary: false,
+    className: ['govuk-input', 'govuk-input--width-10'],
+    validationLink: {
+      field: ''
+    }
+  },
+  'is-legal-representative-email': {
+    isPageHeading: true,
+    mixin: 'radio-group',
+    validate: ['required'],
+    legend: {
+      className: 'visuallyhidden'
+    },
+    options: [{
+      value: 'yes'
+    }, {
+      value: 'no',
+      toggle: 'legal-representative-email-details-fieldset',
+      child: 'partials/legal-representative-email-details'
+    }]
+  },
+  'helper-details': {
+    isPageHeading: true
+  },
+  'someone-else-fullname': {
+    labelClassName: 'bold',
+    mixin: 'input-text',
+    validate: ['required', 'notUrl'],
+    className: ['govuk-input', 'govuk-!-width-two-thirds']
+  },
+  'someone-else-relationship': {
+    labelClassName: 'bold',
+    mixin: 'input-text',
+    validate: ['required', 'notUrl'],
+    className: ['govuk-input', 'govuk-!-width-two-thirds']
+  },
```

Comment:
The validation messages for these two fields are not shown when left blank. They need to be added in to the validation.json file and amended in the html file.

Reply:
ok